crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1650,7 +1651,24 @@ pub(super) fn report_unresolved_reference(context: &InferContext, expr_name_node
     };
 
     let ast::ExprName { id, .. } = expr_name_node;
-    builder.into_diagnostic(format_args!("Name `{id}` used when not defined"));
+    let mut diagnostic = builder.into_diagnostic(format_args!("Name `{id}` used when not defined"));
+    if let Some(version_added_to_builtins) = version_builtin_was_added(id) {
+        diagnostic.info(format_args!(
+            "`{id}` was added as a builtin in Python 3.{version_added_to_builtins}"
+        ));
+
+        // TODO: can we tell the user *why* we're inferring this target version?
+        // CLI flag? pyproject.toml? Python environment?
+        diagnostic.info(format_args!(
+            "The inferred target version of your project is Python {}",
```

Comment:
It seems a bit odd to call it the "inferred" Python version if it came from eg a CLI flag. But I guess this is part of the TODO above?

Reply:
Yes and yes